### PR TITLE
Add voice broadcast pre-recording PiP

### PR DESCRIPTION
**Pull request by weeman1337**
Part of https://github.com/vector-im/element-web/issues/23282

PSF-1686

## Checklist

* [x] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/voice-broadcast/components/atoms/VoiceBroadcastHeader.tsx [hunk 1]
```diff
@@ -19,27 +19,41 @@ import { Icon as LiveIcon } from "../../../../res/img/element-icons/live.svg";
...
         </div>
         : null;
+
     const liveBadge = live ? <LiveBadge /> : null;
+
+    const closeButton = showClose
+        ? <AccessibleButton onClick={onCloseClick}>
+            <XIcon className="mx_Icon mx_Icon_16" />
+        </AccessibleButton>
+        : null;
```
Comment (SimonBrandner) [inline src/voice-broadcast/components/atoms/VoiceBroadcastHeader.tsx:55]: Personally I'd prefer to do these inline like `{ closeButton && <AccessibleButton...>}`

File: src/voice-broadcast/stores/VoiceBroadcastPreRecordingStore.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { TypedEventEmitter } from "matrix-js-sdk/src/models/typed-event-emitter";
+
+import { VoiceBroadcastPreRecording } from "..";
+import { IDestroyable } from "../../utils/IDestroyable";
+
+export type VoiceBroadcastPreRecordingEvent = "changed";
```
Comment (SimonBrandner) [inline src/voice-broadcast/stores/VoiceBroadcastPreRecordingStore.ts:22]: Why not make this an enum?

**Review by SimonBrandner — APPROVED** [review]
LGTM, just some nits

File: src/voice-broadcast/models/VoiceBroadcastPreRecording.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+*/
+
+import { MatrixClient, Room, RoomMember } from "matrix-js-sdk/src/matrix";
+import { TypedEventEmitter } from "matrix-js-sdk/src/models/typed-event-emitter";
+
+import { IDestroyable } from "../../utils/IDestroyable";
+import { VoiceBroadcastRecordingsStore } from "../stores/VoiceBroadcastRecordingsStore";
+import { startNewVoiceBroadcastRecording } from "../utils/startNewVoiceBroadcastRecording";
+
+type VoiceBroadcastPreRecordingEvent = "dismiss";
```
Comment (SimonBrandner) [inline src/voice-broadcast/models/VoiceBroadcastPreRecording.ts:24]: Why not make this an enum?

File: src/voice-broadcast/stores/VoiceBroadcastPreRecordingStore.ts (hunk 2 above)
Comment (weeman1337) [reply src/voice-broadcast/stores/VoiceBroadcastPreRecordingStore.ts:22]: There was a discussion about this in web chapter. The conclusion was to prefer union types… One argument was, that you don't have to import the module if you want to use the value.

**Pull request merged**